src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,320 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import java.util.Random;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static final int MIN_AGE = 18;
+    private static final int MAX_AGE = 100;
+    private static final int NUMBER_OF_USERS = 10;
+    private static final int FIRST_USER_ID = 1;
+    private static final int SECOND_USER_ID = 2;
+
+    private static Random random;
+    private static StorageDao storageDao;
+    private static User userTest;
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    public static void setUp() {
+        random = new Random();
+        userTest = new User();
+        userTest.setAge(MIN_AGE);
+        userTest.setId(30L);
+        userTest.setLogin("CoolGuy");
+        userTest.setPassword("Qwerty123_4");
```

Comment:
no need to set Id, it will be set by the database
if you always use those same setters together, consider creating a constructor